oak-segment-remote/src/main/java/org/apache/jackrabbit/oak/segment/remote/persistentcache/PersistentDiskCache.java

```diff
@@ -193,16 +188,27 @@ public void cleanUp() {
     private void cleanUpInternal() {
         if (isCacheFull()) {
             try {
-                Stream<Path> segmentsPaths = Files.walk(directory.toPath())
+                Stream<SegmentCacheEntry> segmentCacheEntryStream = Files.walk(directory.toPath())
+                        .map(path -> {
+                            try {
+                                return new SegmentCacheEntry(path, Files.readAttributes(path, BasicFileAttributes.class).lastAccessTime());
+                            } catch (IOException e) {
+                                logger.error("Error while getting the last access time for {}", path.toFile().getName());
+                            }
+                            return new SegmentCacheEntry(null, FileTime.fromMillis(Long.MAX_VALUE));
```

Comment:
I think it would be better to return the actual path here. If for some weird reason we cannot read any lastAccessTime we won't be able to clean up the cache at all.
```suggestion
                            return new SegmentCacheEntry(path, FileTime.fromMillis(Long.MAX_VALUE));
```

Reply:
Agree, thanks.